Tools-Override/CloudTest.targets

```diff
@@ -428,10 +428,25 @@
       </ForUpload>
     </ItemGroup>
   </Target>
+  
+  <Target Name="CopyTestILCToRuntimeDir" Condition="'$(BuildingUAPAOTVertical)' == 'true'">
```

Comment:
this target will only come into play if you invoke /t:CloudBuild on tests.builds so if you are running locally it will never execute, meaning that you will still use the ilc from wherever you had it locally. If you do invoke CloudBuild, then we copy it so that it goes with the shared payload.